From fe45ae047a00df1ebd24db2bdd783000a1bb0631 Mon Sep 17 00:00:00 2001
From: Michael Scott <mike@foundries.io>
Date: Wed, 16 Dec 2020 17:43:18 -0800
Subject: [PATCH 041/233] [FIO toup] arch: mach-imx: hab: add SPL_IMX_HAB
 config

This allows granularity of turning off HAB features for SPL
which might not need them.

Signed-off-by: Michael Scott <mike@foundries.io>
Co-developed-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 arch/arm/mach-imx/Kconfig  | 8 ++++++++
 arch/arm/mach-imx/Makefile | 8 ++++----
 arch/arm/mach-imx/spl.c    | 4 ++--
 3 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-imx/Kconfig b/arch/arm/mach-imx/Kconfig
index 0616b05a054..ca3fde380b3 100644
--- a/arch/arm/mach-imx/Kconfig
+++ b/arch/arm/mach-imx/Kconfig
@@ -81,6 +81,14 @@ config IMX_HAB
 	  This option enables the support for secure boot (HAB).
 	  See doc/imx/habv4/* for more details.
 
+config SPL_IMX_HAB
+	bool "Support i.MX HAB features in SPL"
+	default y if IMX_HAB
+	depends on ARCH_MX7 || ARCH_MX6 || ARCH_MX5 || ARCH_MX7ULP || ARCH_IMX8M
+	help
+	  This option enables the support for secure boot (HAB) in SPL.
+	  See doc/README.mxc_hab for more details.
+
 config CSF_SIZE
 	hex "Maximum size for Command Sequence File (CSF) binary"
 	depends on IMX_HAB
diff --git a/arch/arm/mach-imx/Makefile b/arch/arm/mach-imx/Makefile
index 612efd4045a..32c20d48841 100644
--- a/arch/arm/mach-imx/Makefile
+++ b/arch/arm/mach-imx/Makefile
@@ -18,7 +18,7 @@ obj-y += mmc_env.o
 obj-$(CONFIG_FEC_MXC) += mac.o
 obj-$(CONFIG_DWC_ETH_QOS) += mac.o
 obj-$(CONFIG_SYS_I2C_MXC) += i2c-mxv7.o
-obj-$(CONFIG_IMX_HAB) += hab.o
+obj-$(CONFIG_$(SPL_)IMX_HAB) += hab.o
 obj-y += cpu.o
 endif
 
@@ -59,17 +59,17 @@ ifneq ($(CONFIG_SPL_BUILD),y)
 obj-$(CONFIG_IMX_BOOTAUX) += imx_bootaux.o
 endif
 obj-$(CONFIG_SATA) += sata.o
-obj-$(CONFIG_IMX_HAB)    += hab.o
+obj-$(CONFIG_$(SPL_)IMX_HAB)    += hab.o
 obj-$(CONFIG_SYSCOUNTER_TIMER) += syscounter.o
 obj-$(CONFIG_IMX_TRUSTY_OS) += trusty.o
 endif
 ifeq ($(SOC),$(filter $(SOC),mx7ulp))
 obj-y  += cache.o mmdc_size.o
-obj-$(CONFIG_IMX_HAB) += hab.o
+obj-$(CONFIG_$(SPL_)IMX_HAB) += hab.o
 endif
 ifneq ($(CONFIG_SPL_BUILD),y)
 ifeq ($(SOC),$(filter $(SOC),mx6 mx7ulp imx8m))
-obj-$(CONFIG_IMX_HAB) += fiohab.o
+obj-$(CONFIG_$(SPL_)IMX_HAB) += fiohab.o
 endif
 ifeq ($(SOC),$(filter $(SOC),imx8))
 obj-$(CONFIG_AHAB_BOOT) += fiohab.o
diff --git a/arch/arm/mach-imx/spl.c b/arch/arm/mach-imx/spl.c
index 207b769a3dc..9ad366c687b 100644
--- a/arch/arm/mach-imx/spl.c
+++ b/arch/arm/mach-imx/spl.c
@@ -257,7 +257,7 @@ u32 spl_mmc_boot_mode(const u32 boot_device)
 }
 #endif
 
-#if defined(CONFIG_IMX_HAB)
+#if defined(CONFIG_SPL_IMX_HAB)
 
 /*
  * +------------+  0x0 (DDR_UIMAGE_START) -
@@ -449,7 +449,7 @@ exit:
 
 void board_spl_fit_post_load(const void *fit, struct spl_image_info *spl_image)
 {
-	if (IS_ENABLED(CONFIG_IMX_HAB) && !(spl_image->flags & SPL_FIT_BYPASS_POST_LOAD)) {
+	if (CONFIG_IS_ENABLED(IMX_HAB) && !(spl_image->flags & SPL_FIT_BYPASS_POST_LOAD)) {
 		u32 offset = ALIGN(fdt_totalsize(fit), 0x1000);
 
 		if (imx_hab_authenticate_image((uintptr_t)fit,
-- 
2.40.1

